dev/reference/tidb-binlog/faq.md

```diff
@@ -115,9 +115,96 @@ Drainer 启动的时候会去读取 checkpoint，如果读取不到，就会使
 
 TiDB 配置开启 `ignore-error` 写 binlog 失败后触发 critical error 告警，后续都不会再写 binlog，所以会有 binlog 数据丢失。如果要恢复同步，需要如下处理：
 
-1. 停止当前 drainer。
+1. 停止当前 Drainer。
 2. 重启触发 critical error 的 `tidb-server` 实例重新开始写 binlog（触发 critical error 后不会再写 binlog 到 pump）。
 3. 上游做全量备份。
 4. 清理掉下游数据包括 checkpoint 表 `tidb_binlog.checkpoint`。
 5. 使用上游的全量备份恢复下游。
-6. 部署 drainer，使用 `initialCommitTs`= {从全量备份获取快照的时间戳}。
+6. 部署 Drainer，使用 `initialCommitTs`= {从全量备份获取快照的时间戳}。
+
+## 在什么情况下暂停和下线 Pump/Drainer？
+
+首先需要通过以下内容来了解 Pump/Drainer 的状态定义和启动、退出的流程。
+
+暂停主要针对临时需要停止服务的场景，例如：
+
+- 版本升级：停止进程后使用新的 binary 启动服务。
+- 服务器维护：需要对服务器进行停机维护。退出进程，等维护完成后重启服务。
+
+下线主要针对永久（或长时间）不再使用该服务的场景，例如：
+
+- Pump 缩容：不再需要那么多 Pump 服务了，所以下线部分服务。
+- 同步任务取消：不再需要将数据同步到某个下游，所以下线对应的 Drainer。
+- 服务器迁移：需要将服务迁移到其他服务器。下线服务，在新的服务器上重新部署。
+
+## 可以通过哪些方式暂停 Pump/Drainer？
+
+- 直接 kill 进程。
+
+    >**注意：**
+    >
+    > 不能使用 kill -9，这种情况 Pump/Drainer 无法对信号进行处理
+
+- 如果 Pump/Drainer 运行在前台，则可以通过按下 <kbd>Ctrl</kbd>+<kbd>C</kbd> 来暂停。
+- 使用 binlogctl 的 `pause-pump` 或 `pause-drainer` 命令。
+
+## 可以使用 binlogctl 的 `update-pump`/`update-drainer` 命令来暂停 Pump/Drainer 服务吗？
+
+不可以。`update-pump`/`update-drainer` 命令直接修改 PD 中保存的状态信息。执行这个命令并不会通知 Pump/Drainer 做相应的操作，**而且使用不当会使数据同步中断，甚至造成数据丢失。**
+
+## 什么情况下使用 binlogctl 的 `update-pump` 命令设置 Pump 状态为 `paused`？
+
+在某些异常情况下，Pump 没有正确维护自己的状态，实际上状态应该为 `paused`。这时可以使用 `update-pump` 对状态进行修正，例如：
+
+- Pump 异常退出（可能由 panic 或者误操作执行 kill -9 直接 kill 掉进程导致），Pump 保存在 PD 中的状态仍然为 `online`。如果暂时不需要重启 Pump 恢复服务，可以使用 `update-pump` 把该 Pump 状态设置为 `paused`，避免对 TiDB 写 binlog 和 Drainer 获取 binlog 造成干扰。
+
+## 什么情况下使用 binlogctl 的 `update-drainer` 命令设置 Drainer 状态为 `paused`？
+
+在某些异常情况下，Drainer 没有正确维护自己的状态，，对数据同步造成了影响，实际上状态应该为 `paused`。这时可以使用 `update-drainer` 对状态进行修正，例如：
+
+- Drainer 异常退出（出现 panic 直接退出进程，或者误操作执行 kill -9 直接 kill 掉进程），Drainer 保存在 PD 中的状态仍然为 `online`。当 Pump 启动时无法正常通知该 Drainer（报错 `notify drainer ...`），导致 Pump 无法正常运行。这个时候可以使用 `update-drainer` 将 Drainer 状态更新为 `paused`，再启动 Pump。
+
+## 可以通过哪些方式下线 Pump/Drainer？
+
+目前只可以使用 binlogctl 的 `offline-pump` 和 `offline-drainer` 命令来下线 Pump 和 Drainer。
+
+## 可以使用 binlogctl 的 `update-pump`/`update-drainer` 命令来下线 Pump/Drainer 服务吗？
```

Comment:
Would it be better to move this Q&A to Line 151?

Reply:
I agree.